interp/src/debugger/cidr.rs

```diff
@@ -122,15 +131,16 @@ impl Debugger {
             &config,
         )?;
 
-        Debugger::new(&components, main_component, None, env)
+        Debugger::new(&components, main_component, None, env, Some(mapping))
     }
 
     pub fn new(
         context: &iir::ComponentCtx,
         main_component: &Rc<iir::Component>,
         source_map: Option<SourceMap>,
         env: InterpreterState,
-    ) -> InterpreterResult<Self> {
+        metadata: Option<NewSourceMap>,
+    ) -> InterpreterResult<(Self, NewSourceMap)> {
```

Comment:
I think we can leave this constructor only returning a debugger for the time
being, and have the `from_file` one be used when there is a need to return the
metadata table. There's also not much point in passing in the metadata map only
to return it as a non-option since this will break the code in places where the
metadata map isn't present